src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py

```diff
@@ -545,6 +545,7 @@ def __call__(
         height: Optional[int] = None,
         width: Optional[int] = None,
         num_inference_steps: int = 50,
+        max_inference_steps: Optional[int] = None,
```

Comment:
I don't fully understand why we need to add this to text-to-image? Why not just use `callbacks` to return unfinished images?